osu.Game/Screens/OnlinePlay/Lounge/Components/DrawableRoom.cs

```diff
@@ -359,63 +240,40 @@ protected override void LoadComplete()
                     specialCategoryPill.Hide();
             }, true);
 
+            roomType.BindTo(Room.Type);
+            roomType.BindValueChanged(t =>
+            {
+                endDateInfo.Alpha = t.NewValue == MatchType.Playlists ? 1 : 0;
+            }, true);
+
             hasPassword.BindTo(Room.HasPassword);
             hasPassword.BindValueChanged(v => passwordIcon.Alpha = v.NewValue ? 1 : 0, true);
         }
 
-        public Popover GetPopover() => new PasswordEntryPopover(Room) { JoinRequested = lounge.Join };
-
-        public MenuItem[] ContextMenuItems => new MenuItem[]
-        {
-            new OsuMenuItem("Create copy", MenuItemType.Standard, () =>
-            {
-                lounge?.Open(Room.DeepClone());
-            })
-        };
-
-        public bool OnPressed(GlobalAction action)
+        protected override IReadOnlyDependencyContainer CreateChildDependencies(IReadOnlyDependencyContainer parent)
         {
-            if (selectedRoom.Value != Room)
-                return false;
-
-            switch (action)
+            return new CachedModelDependencyContainer<Room>(base.CreateChildDependencies(parent))
             {
-                case GlobalAction.Select:
-                    TriggerClick();
-                    return true;
-            }
-
-            return false;
-        }
-
-        public void OnReleased(GlobalAction action)
-        {
+                Model = { Value = Room }
+            };
         }
 
-        protected override bool ShouldBeConsideredForInput(Drawable child) => state == SelectionState.Selected || child is HoverSounds;
+        private int numberOfAvatars = 7;
 
-        protected override bool OnClick(ClickEvent e)
+        public int NumberOfAvatars
```

Comment:
Not going to dwell on this, but we usually have public properties at the top right?

Reply:
These days I put business-logic properties below the ctor, because when you have too many of them both the ctor and `load()` get pushed down and generally I'm more interested in looking at those if I'm working with a class.

My typical structure is something like:
```
1. Public consts
2. Private consts
3. Public auto/inline properties.
4. Public readonly fields.
5. Public fields.
6. Private resolved dependencies.
7. Private readonly fields.
8. Private fields
9. Ctor
10. load()
11. LoadComplete()
12. Public complex properties.
13. Whatever else (no real structure here atm, typically public methods at the top and privates at the bottom, but this doesn't always make sense).
14. (At the end of the file) Abstract methods
```